src/Store/Query.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Patchlevel\EventSourcing\Store;
+
+use function array_merge;
+use function array_values;
+
+/** @experimental */
+final class Query
+{
+    /** @var list<SubQuery> */
+    public readonly array $subQueries;
+
+    public function __construct(
+        SubQuery ...$subQueries,
+    ) {
+        $this->subQueries = array_values($subQueries);
+    }
+
+    public function add(SubQuery $subQuery): self
+    {
+        return new self(...array_merge($this->subQueries, [$subQuery]));
```

Comment:
```suggestion
        return new self([...$this->subQueries, $subQuery]);
```
Should this be sufficient and maybe a little bit more performant?